src/services/dashboards/widgets/cost-map/CostMapWidget.vue

```diff
@@ -69,40 +67,71 @@ interface CostMapData {
 const props = defineProps<WidgetProps>();
 
 const chartContext = ref<HTMLElement | null>(null);
-const {
-    createTreeMapSeries,
-    createTooltip, setTreemapTooltipText,
-    disposeRoot, refreshRoot,
-} = useAmcharts5(chartContext);
+const chartHelper = useAmcharts5(chartContext);
 
 const state = reactive({
-    ...toRefs(useWidgetState<CostMapData>(props)),
-    chartData: computed(() => {
-        if (!state.data) return [];
-        return state.data;
+    ...toRefs(useWidgetState<HistoryDataModel['results']>(props)),
+    groupBy: computed<GroupBy>(() => state.options.group_by ?? GROUP_BY.PROJECT),
+    chartData: computed(() => getRefinedTreemapChartData(state.data, state.groupBy, props.allReferenceTypeInfo)),
+    dateRange: computed<DateRange>(() => {
+        const end = state.settings?.date_range?.end ?? dayjs.utc().format('YYYY-MM');
+        const start = state.settings?.date_range?.start ?? dayjs.utc(end).subtract(11, 'month').format('YYYY-MM');
+        return { start, end };
     }),
+    // Todo: Link to target
+    // widgetLink: computed(() => ({
+    //     name: COST_EXPLORER_ROUTE.COST_ANALYSIS._NAME,
+    //     params: {},
+    //     query: {
+    //         granularity: primitiveToQueryString(GRANULARITY.ACCUMULATED),
+    //         groupBy: arrayToQueryString([state.groupBy]),
+    //         period: objectToQueryString(state.dateRange),
+    //     },
+    // })),
 });
 
 const widgetFrameProps:ComputedRef = useWidgetFrameProps(props, state);
 
 /* Api */
-const fetchData = async () => new Promise((resolve) => {
-    setTimeout(() => {
-        resolve(SAMPLE_RAW_DATA);
-    }, 1000);
-});
+const fetchData = async (): Promise<TreemapChartData[]> => {
+    try {
+        const apiQueryHelper = new ApiQueryHelper();
+        if (state.groupBy === 'project_id') apiQueryHelper.setFilters([{ k: 'project_id', v: null, o: '!=' }]);
```

Comment:
todo: `groupBy` of this widget may be dynamical!

```suggestion
        if (state.groupBy === state.groupBy) apiQueryHelper.setFilters([{ k: state.groupBy, v: null, o: '!=' }]);
```